server/neptune/workflows/internal/metrics/handler.go

```diff
@@ -0,0 +1,66 @@
+package metrics
+
+import (
+	"strings"
+
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/workflow"
+)
+
+const delim = "."
+
+// Scope is an interface that attempts to wrap temporal's MetricsHandler with additional
+// functionality to build namespaces.
+// Namespaces are of the following form:
+// <s1>.<s2>.<s3> and can have a number of tags associated with it.
+type Scope interface {
+	SubScope(namespace ...string) Scope
+	SubScopeWithTags(tags map[string]string) Scope
```

Comment:
nit: should this method be just `WithTags()` since we're not adding a new sub scope here?

Reply:
it technically is a new subscope, we're creating a new scope, it's just not a new namespace.  Just wanted to keep this consistent.